server/log_rpc_server.go

```diff
@@ -96,8 +100,11 @@ func (t *TrillianLogRPCServer) QueueLeaves(ctx context.Context, req *trillian.Qu
 	}
 	ctx = trees.NewContext(ctx, tree)
 
-	for i := range req.Leaves {
-		req.Leaves[i].MerkleLeafHash = hasher.HashLeaf(req.Leaves[i].LeafValue)
+	for _, leaf := range req.Leaves {
+		leaf.MerkleLeafHash = hasher.HashLeaf(leaf.LeafValue)
+		if len(leaf.LeafIdentityHash) == 0 {
```

Comment:
Is there a way we could document this behavior somewhere? Perhaps a comment on the proto or rpc definition? 

In the [log client](https://github.com/google/trillian/blob/master/client/verifier.go#L100), we're using SHA256 to fill `LeafIdentityHash`.  It would be nice to have these two be consistent.

Reply:
Done and done. I found a few other (now) redundant usages of LeafIdentityHash and removed those too.